### PR TITLE
Add C++ tutorial and template app information

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-39432

- [C++ - Device Sync](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-39432/tutorial/cpp/): New C++ tutorial based on the template app. Note this is based on the [template app PR](https://github.com/mongodb-university/realm-template-apps/pull/206) that is currently waiting for review.
- [Get Started Landing Page](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-39432/tutorials/): Add a card for the C++ tutorial.
- [Template Apps](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-39432/reference/template-apps/): Add the C++ template app ID and link to the GitHub repository. Minor procedure updates for UI changes, and minor naming tweaks.

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Did you tag pages appropriately?
  - genre
  - programming_language
  - meta.keywords
  - meta.description
- [ ] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-realm work, if any

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/tutorial/cpp.txt [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+              };
+
+              _database->write([&]{
+                _database->add(std::move(item));
+              });
+            }
+
+   .. step:: Run and Test
+    
+      At this point, you can run the application again. Log in using the account
```
Comment (cbullinger) [inline source/tutorial/cpp.txt:510]: I had to rebuild before running. not sure if that's not the intended workflow or if it's just an assumed step to "run the application", but it tripped me up for a second when I didn't see my changes

File: source/tutorial/cpp.txt [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+      :guilabel:`Create App from Template`. Select the 
+      :guilabel:`Real-time Sync` template. This creates an App Services App
+      pre-configured to use with one of the Device Sync template app clients.
+
+      After you create a template app, the UI displays a modal labeled 
+      :guilabel:`Get the Front-end Code for your Template`. This modal 
+      provides instructions for downloading the template app client code 
+      as a ``.zip`` file or using {+cli+} to get the client.
+
+      The C++ Template App is not yet available to download in the App
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/tutorial/cpp.txt:102]: You can still use the UI to create a template app, and that sets up the backend for you. You just can't download the C++ client from the UI yet, although there is a Jira ticket to add this so hopefully that will change soon.

File: source/tutorial/cpp.txt [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+   .. tab:: GitHub Repository
+      :tabid: github
+
+      You can fork and clone a GitHub repository that contains the Device 
+      Sync client code. The C++ client code is available at
+      :github:`https://github.com/mongodb/template-app-cpp-todo <mongodb/template-app-cpp-todo>`.
+
+      If you use this process to get the client code, you must create a
+      template app to use with the client. Follow the instructions at 
+      :ref:`create-template-app` to use the Atlas App Services UI, {+cli+},
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/tutorial/cpp.txt:138]: No, it's still possible to use the UI to create the template app, which configures the backend - you just can't get the client code from it.

File: source/tutorial/cpp.txt [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+         The top of the home screen contains a row of buttons and a toggle to
+         hide completed tasks. If the terminal window is small, the buttons
+         display as rectangles too small to display the text labels. If you
+         don't see the text labels, make the terminal window bigger. FTXUI
+         re-renders the content to fit the larger window.
+
+   .. step:: Check the Backend
+
+      Log in to `Atlas App Services <http://cloud.mongodb.com>`_. In the 
+      :guilabel:`Data Services` tab, click on :guilabel:`Browse Collections`. In the list
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/tutorial/cpp.txt:278]: It's the same thing. But I think we used `Data Services` here to differentiate between it and `App Services`, which the user may already be in if they just created a template app.

I actually don't remember if this was in Caleb's original template app tutorial, or if I added it somewhere along the way. But this is the language from the SwiftUI tutorial.

File: source/tutorial/cpp.txt [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+      
+      In a production app, you might add a ``PriorityLevel`` enum to constrain
+      the possible values. For this tutorial, we'll use a number property to
+      simplify working with the UI framework.
+      
+      To do this, follow these steps:
+
+      A. Open the client code in your preferred IDE. 
+
+      #. Open the ``item.hpp`` file in the ``state/`` directory.
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/tutorial/cpp.txt:305]: When writing tutorial steps, I generally prefer to start with the action and then provide any supporting details. That gives the user the most important contextual info first - what am I doing - and then the info about how to do it. I can't remember if it was a former editor or a style guide that got me into this habit, but it has stuck with me and I generally think it's the right way to structure content for tutorials.

File: source/tutorial/cpp.txt (hunk 1 above)
Comment (dacharyc) [reply source/tutorial/cpp.txt:510]: It's a miss on my part - in Xcode/SwiftUI, running the app builds it with the latest changes. I forgot to update this for C++. I added the info here.

**Pull request merged**